federation/cmd/federation-controller-manager/app/controllermanager.go

```diff
@@ -114,6 +125,15 @@ func Run(s *options.CMServer) error {
 		glog.Fatal(server.ListenAndServe())
 	}()
 
+	federationClientset, err := federationclientset.NewForConfig(restclient.AddUserAgent(restClientCfg, "federation-controller-manager"))
+	if err != nil {
+		glog.Fatalf("Invalid API configuration: %v", err)
+	}
+
+	if err := createFederationNamespace(federationClientset, federationOnlyNamespace); err != nil {
```

Comment:
You should only do this if the user has requested an HA federation control plane, otherwise you have an unnecessary namespace hanging around, for no reason.